lib/elixir/test/elixir/kernel/impl_test.exs

```diff
@@ -495,6 +495,16 @@ defmodule Kernel.ImplTest do
              "got \"@impl Kernel.ImplTest.MacroBehaviour\" for function foo/0 but this behaviour does not specify such callback"
   end
 
+  test "raises when :to targeting the delegating module is given without the :as option" do
+    assert_raise ArgumentError,
+                 ~r/to: cannot target the delegating module without an as: alias/,
+                 fn ->
+                   defmodule ImplAttributes do
+                     defdelegate foo(), to: __MODULE__
+                   end
+                 end
+  end
+
```

Comment:
This test should be moved to `kernel_test.exs`, as it is not related to the `@impl` attribute.